util/codecov-ci.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+#
+# Copyright Amazon.com Inc. or its affiliates.  All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+#
+
+set -xe
+
+SRC=$(pwd)
+
+# Sanity check
+DIRNAME=$(basename -- "${SRC}")
+if [[ "${DIRNAME}" != 'aws-lc' ]]; then
+  echo "Script must be executed from aws-lc directory"
+  exit 1
+fi
+
+BUILD="$1"
+if [ -n "$BUILD" ]; then
+  if [ ! -e "$BUILD" ]; then
+    mkdir -p "$BUILD"
+  fi
+  BUILD=$(readlink -f "$BUILD")
+else
+  echo "Must specify a build directory."
+  exit 1
+fi
+
+cmake -DGCOV=1 -DDISABLE_PERL=1 -DBUILD_TESTING=1 -DBUILD_LIBSSL=1 -DCMAKE_BUILD_TYPE=Debug -S "${SRC}" -B "${BUILD}"
```

Comment:
do we need debug symbols for coverage? should we consider `-DCMAKE_BUILD_TYPE=Release`? or `-DCMAKE_BUILD_TYPE=RelWithDebInfo`?

Reply:
I chose `Debug` to avoid compiler optimizations that might obscure the coverage metrics.